### PR TITLE
Refactor search and query into one

### PR DESCRIPTION
**Pull request by robert-fjolnir**

### CODE REVIEW

**Review by gasparyanartur — APPROVED** [review]
Nice!

**Comment by danewhitfield** [comment]
Looks great :)

File: src/repositories/library-logic.js [hunk 1]
```diff
@@ -84,6 +84,7 @@ export const addComponentToDatabase = async ({ name, prompt, embedding, componen
         embedding: blob,
         prompt: prompt,
         component: component,
```
Comment (danewhitfield) [inline src/repositories/library-logic.js:86]: Looks perfect, thanks!

File: src/routes/library.js [hunk 2]
```diff
@@ -142,9 +144,9 @@ router.post(':id/downvote', async (req, res, next) => {
 });
 
 router.post('/test', async (req, res) => {
-    await addComponentToDatabase({ name: 'a', prompt: 'foo foo a', embedding: [0.02, 0.59], component: 'hello' }),
-        await addComponentToDatabase({ name: 'b', prompt: 'bar bar b', embedding: [0.03, 0.14], component: 'my' }),
-        await addComponentToDatabase({ name: 'c', prompt: 'baz baz c', embedding: [0.6, 0.4], component: 'name' }),
```
Comment (danewhitfield) [inline src/routes/library.js:140]: Awesome!

**Review by gasparyanartur — APPROVED** [review]

**Pull request merged**